File: app/src/main/java/org/techtown/moback/MyViewHolder.java
package org.techtown.moback;

import android.view.View;

import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;
import androidx.recyclerview.widget.RecyclerView;

public class MyViewHolder<T extends ViewDataBinding> extends RecyclerView.ViewHolder {


    private final T binding;

    public MyViewHolder(final View v){
        super(v);
        this.binding = (T) DataBindingUtil.bind(v);

    }

    public T binding() {
        return binding;
    }
}

File: app/src/main/java/org/techtown/moback/MyViewHolder.kt
package org.techtown.moback

import android.view.View
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import androidx.recyclerview.widget.RecyclerView

class MyViewHolder<T : ViewDataBinding>(v: View) : RecyclerView.ViewHolder(v) {
    private val binding: T
    fun binding(): T {
        return binding
    }

    init {
        binding = DataBindingUtil.bind<ViewDataBinding>(v) as T
    }
}